### ISSUE TITLE
Feature: Timeout override per backend

### ISSUE THREAD

**Issue by sirlatrom** [containous/traefik#3027]
### Do you want to request a *feature* or report a *bug*?
Feature: The ability to specify the various timeouts as part of configuring a backend.

### What did you do?
I searched through the documentation, the issues on GitHub and asked in Slack. AFAICT, it is currently only possible to specify timeouts globally.

My use case is to specify reasonable default/global timeout values, but being able to override them to either cut connections to backend servers which sometimes are too slow to respond, or to allow long-running requests to complete.

### What did you expect to see?
Something like what's available in HAProxy, e.g. the [timeout server](https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#4-timeout%20server) configuration, which is specifiable both globally and per backend.

### What did you see instead?
[Globally configured timeouts only](https://docs.traefik.io/v1.5/configuration/commons/#timeouts).

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.5.3
Codename:     cancoillotte
Go version:   go1.9.4
Built:        2018-02-27_02:47:04PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
N/A.
### If applicable, please paste the log output in debug mode (`--debug` switch)
N/A.

**Comment by thiago** [comment]
When I have many applications behind traefik, it really makes sense to be able to configure timeouts per backend, since global timeouts do not make sense to all applications. I think this is applicable for "Responding Timeouts" and "Forwarding Timeouts".

**Comment by sese** [comment]
I'd like to have this feature, On production, I use haproxy for a WebSocket application, after 15 minutes we have to disconnect the client and show a proper message. Now we test this on a production, because we cannot test it on a machine with traefik. So please be sure you add something similar:

> timeout tunnel <timeout>

[haproxy](https://cbonte.github.io/haproxy-dconv/1.7/configuration.html#4-timeout%20tunnel)

**Comment by pondohva** [comment]
Also, I need to specify this timeouts per backend from kubernetes ingress annotations

**Comment by acomisario** [comment]
It's really weird that traefik doesnt support this, and we need this also, sadly everytime something like this is needed, we need to take the frontend out of traefik :(

**Comment by BcTpe4HbIu** [comment]
Any chance for this feature to get implemented? Or any reason why it should not be?

And there is possible duplicate: #3671

**Comment by sergeypavlikhin** [comment]
Any updates? 
We need this too :(

**Comment by umomany** [comment]
+1

**Comment by debu99** [comment]
we need it too

**Comment by notsureifkevin** [comment]
@sirlatrom it is now possible to configure timeouts for each entrypoint in v2

https://docs.traefik.io/routing/entrypoints/#transport

**Comment by sirlatrom** [comment]
@notsureifkevin Thanks for the heads up! I guess that's an improvement over v1, but still doesn't meet the need that some have to control per ~backend~ service/(router?), perhaps by middleware, the timeouts more granularly.

**Comment by sirlatrom** [comment]
I suppose https://docs.traefik.io/middlewares/circuitbreaker/#latencyatquantilems could be used to yield a timeout-like effect at the aggregate level, but it doesn't provide for hard and fast timeouts.

**Comment by notsureifkevin** [comment]
> 
> 
> @notsureifkevin Thanks for the heads up! I guess that's an improvement over v1, but still doesn't meet the need that some have to control per ~backend~ service/(router?), perhaps by middleware, the timeouts more granularly.

That is true, however, you could define an entrypoint for each service that you wanted to deviate from the default timeouts - this in effect is even more flexible than say an ALB provided by AWS, where they only allow setting the timeout on the load balancer itself.

**Comment by sirlatrom** [comment]
> > 
> > 
> > @notsureifkevin Thanks for the heads up! I guess that's an improvement over v1, but still doesn't meet the need that some have to control per ~backend~ service/(router?), perhaps by middleware, the timeouts more granularly.
> 
> That is true, however, you could define an entrypoint for each service that you wanted to deviate from the default timeouts - this in effect is even more flexible than say an ALB provided by AWS, where they only allow setting the timeout on the load balancer itself.

Agreed, however when originally creating this issue we were coming from HAProxy where we could already configure timeouts per backend. With an entrypoint per deviating service we'd also need to hit different ports as they cannot be reused across entrypoints, and it wouldn't be very flexible if we needed to suddenly restart traefik just to set a timeout for a specific service.

**Comment by michael-todorovic** [comment]
We need this too :+1:
